### PR TITLE
Mobile provider mode

### PR DESCRIPTION
**Pull request by Zensey**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -146,6 +147,7 @@ func (cfg *Config) GetConfig() map[string]interface{} {
 
 // SetDefault sets default value for key.
 func (cfg *Config) SetDefault(key string, value interface{}) {
+	fmt.Println("!SetDefault >", key, value)
```
Comment (n10ty) [inline config/config.go:150]: Some local debug?

File: mobile/mysterium/entrypoint.go [hunk 2]
```diff
@@ -716,3 +745,56 @@ func (mb *MobileNode) HealthCheck() *HealthCheckData {
...
+	chainID := config.GetInt64(config.FlagChainID)
+	id, err := mb.identitySelector.UseOrCreate(adr, passphrase, chainID)
+	if err != nil {
+		return ""
+	}
+	return id.Address
+}
+
+func (mb *MobileNode) StartProvider() {
+	fmt.Println("StartProvider>")
```
Comment (n10ty) [inline mobile/mysterium/entrypoint.go:759]: Same here

File: mobile/mysterium/entrypoint.go [hunk 3]
```diff
@@ -716,3 +745,56 @@ func (mb *MobileNode) HealthCheck() *HealthCheckData {
...
+	serviceTypes := strings.Split(activeServices, ",")
+
+	for _, serviceType := range serviceTypes {
+		serviceOpts, err := services.GetStartOptions(serviceType)
+		if err != nil {
+			return
+		}
+
+		id, err := mb.servicesManager.Start(identity.Identity{Address: providerID}, serviceType, serviceOpts.AccessPolicyList, serviceOpts.TypeOptions)
+		if err != nil {
```
Comment (n10ty) [inline mobile/mysterium/entrypoint.go:777]: Same, would be nice to log error

File: mobile/mysterium/entrypoint.go [hunk 4]
```diff
@@ -716,3 +745,56 @@ func (mb *MobileNode) HealthCheck() *HealthCheckData {
...
+			return
+		}
+		mb.serviceIDs = append(mb.serviceIDs, id)
+	}
+}
+
+func (mb *MobileNode) StopProvider() {
+	fmt.Println("StopProvider>")
```
Comment (n10ty) [inline mobile/mysterium/entrypoint.go:785]: Debug?

File: config/config.go (hunk 1 above)
Comment (Zensey) [reply config/config.go]: OK

File: mobile/mysterium/entrypoint.go [hunk 5]
```diff
@@ -716,3 +745,56 @@ func (mb *MobileNode) HealthCheck() *HealthCheckData {
...
+		config.FlagIdentity.Value,
+		config.FlagIdentityPassphrase.Value,
+	)
+	log.Info().Msgf("Unlocked identity: %v", providerID)
+
+	activeServices := config.Current.GetString(config.FlagActiveServices.Name)
+	serviceTypes := strings.Split(activeServices, ",")
+
+	for _, serviceType := range serviceTypes {
+		serviceOpts, err := services.GetStartOptions(serviceType)
```
[the remark this answers is not in the document]
Comment (Zensey) [reply mobile/mysterium/entrypoint.go:770]: OK

File: mobile/mysterium/entrypoint.go (hunk 3 above)
Comment (Zensey) [reply mobile/mysterium/entrypoint.go:777]: right

File: mobile/mysterium/entrypoint.go (hunk 4 above)
Comment (Zensey) [reply mobile/mysterium/entrypoint.go]: OK. removed

File: mobile/mysterium/entrypoint.go [hunk 6]
```diff
@@ -298,16 +319,19 @@ func NewNode(appPath string, options *MobileNodeOptions) (*MobileNode, error) {
...
 				HermesID:           options.HermesSCAddress,
 				ChannelImplAddress: options.ChannelImplementationSCAddress,
 				MystAddress:        options.MystSCAddress,
 				ChainID:            options.Chain2ID,
+				KnownHermeses:      config.GetStringSlice(config.FlagChain2KnownHermeses),
 			},
 		},
-		Consumer:        true,
+
+		Consumer:        false,
```
Comment (soffokl) [inline mobile/mysterium/entrypoint.go:334]: @Zensey if we merge this, will it break our current mobile app? 
Feels like it should be in a separate branch to keep our mobile builds alive.

Comment (Zensey) [reply mobile/mysterium/entrypoint.go:334]: @soffokl 
as long as it was decided to make a separate app for provider, I'll make a separate function

**Pull request closed without merging**